### PR TITLE
add golangci lint to github action workflow

### PR DESCRIPTION
**Pull request by suchen-sci**
- rm unused files, variables, consts
- rm unnecessary type assert for switch
- some are deprecated, need to update
- other unnecessary operations

### CODE REVIEW

File: pkg/protocols/httpprot/httpstat/httpstat.go [hunk 1]
```diff
@@ -46,10 +46,6 @@ type (
 		errRate5  metrics.EWMA
 		errRate15 metrics.EWMA
 
-		m1ErrPercent  float64
```
Comment (suchen-sci) [inline pkg/protocols/httpprot/httpstat/httpstat.go:49]: not used.

File: pkg/supervisor/mock.go [hunk 2]
```diff
@@ -18,26 +18,21 @@
...
 
 import (
-	"sync"
-
 	"github.com/megaease/easegress/v2/pkg/cluster"
 	"github.com/megaease/easegress/v2/pkg/option"
 )
 
 // NewMock return a mock supervisor for testing purpose
-func NewMock(options *option.Options, cls cluster.Cluster, businessControllers sync.Map,
```
Comment (suchen-sci) [inline pkg/supervisor/mock.go:28]: remove `sync.Map`.

File: pkg/util/jmxtool/common.go [hunk 3]
```diff
@@ -52,25 +52,24 @@ func JSONToKVMap(jsonStr string) (map[string]string, error) {
 func extractKVs(prefix string, obj interface{}) []map[string]string {
 	var rst []map[string]string
 
-	switch obj.(type) {
```
Comment (suchen-sci) [inline pkg/util/jmxtool/common.go:55]: unnecessary type assertion.

File: pkg/util/sem/semaphore_test.go [hunk 4]
```diff
@@ -68,7 +68,8 @@ func runCase(s *Semaphore, maxCount int64, t *testing.T) {
 	wg.Wait()
 
 	// step 2: try to acquire one more, should timeout
-	ctx, _ := context.WithTimeout(context.Background(), 100*time.Millisecond)
```
Comment (suchen-sci) [inline pkg/util/sem/semaphore_test.go:71]: not call `cancel`.

File: pkg/util/signer/signer.go [hunk 5]
```diff
@@ -50,53 +50,53 @@ type (
 	Literal struct {
 		// ScopeSuffix is the last part when build the credential scope.
 		// Default: megaease_request
-		ScopeSuffix string `json:"scopeSuffix" json:"scopeSuffix" jsonschema:"required"`
```
Comment (suchen-sci) [inline pkg/util/signer/signer.go:53]: `json` twice.

**Review by xxx7xxxx — APPROVED** [review]

**Pull request merged**